core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -408,14 +408,17 @@ class TaskManager(taskManagerResources: SystemResources = TaskManagerDefaults.de
         }
         false
       case _ =>
+        logger.debug(f"3 ${node.task.name} : ${tasks.map(_.name).mkString(", ")}")
         // set the submission time stamp
-        taskExecutionInfoFor(node).foreach(info => info.submissionDate = Some(Instant.now()))
+        node.taskInfo.submissionDate = Some(Instant.now())
         // we will make this task dependent on the tasks it creates...
         if (tasks.contains(node.task)) throw new IllegalStateException(s"Task [${node.task.name}] contained itself in the list returned by getTasks")
         // track the new tasks. If they are already added, that's fine too.
-        val taskIds: List[TaskId] = for (task <- tasks) yield addTask(task = task, enclosingNode = Some(node), ignoreExists = true)
+        val taskIds: List[TaskId] = tasks.map { task => addTask(task = task, enclosingNode = Some(node), ignoreExists = true) }
         // make this node dependent on those tasks
         taskIds.map(taskId => node.addPredecessors(this(taskId)))
+        // we may need to update precedessors if a returned task was already completed
+        if (tasks.flatMap(t => graphNodeFor(t)).exists(_.state == GraphNodeState.COMPLETED)) updatePredecessors()
```

Comment:
2nd bug (2a): if a completed task is returned by `getTasks`, the dependency graph may not be updated correctly